### ISSUE TITLE
"cannot identify image file" in 3.0.0 on a TIFF file that worked with 2.9.0

### ISSUE THREAD

**Issue by pawl** [python-pillow/Pillow#1466]
To test, first download the files in this folder: https://github.com/flask-admin/flask-admin/tree/master/flask_admin/tests/data

Then run this code in the same folder as those files:
```
from PIL import Image

file_list = [
    'copyleft.gif',
    'copyleft.jpeg',
    'copyleft.jpg',
    'copyleft.png',
    'copyleft.tiff',
]

for filename in file_list:
    with open(filename, 'rb') as fp:
        image = Image.open(fp)
        print("successfully opened:", filename)
```

It will open all 5 files in 2.9.0, but in 3.0.0 it will fail on the TIFF file. It's failing on OS X Yosemite and Ubuntu 12.04 LTS Server Edition 64 bit with both Python 2 and 3.

The error I'm getting is: ```IOError: cannot identify image file <open file 'copyleft.tiff', mode 'rb' at 0x10a15f8a0>```

**Comment by wiredfool** [comment]
Well, It's the Tiff IFD change.

**Comment by wiredfool** [comment]
The tiff spec defines SAMPLEFORMAT as N shorts, where N is the number of bands in the image. 

For the old code, the sample format was set by getscalar(SAMPLEFORMAT, (1,)), where getformat was defined as:
```
    def getscalar(self, tag, default=None):
        try:
            value = self[tag]
            if len(value) != 1:
                if tag == SAMPLEFORMAT:
                    # work around broken (?) matrox library
                    # (from Ted Wright, via Bob Klimek)
                    raise KeyError  # use default
                raise ValueError("not a scalar")
            return value[0]
        except KeyError:
            if default is None:
                raise
            return default
```

Which I think simply means that `SAMPLEFORMAT` isn't a scalar, and would always return the default `(1,)` value unless we happened to have a single band image, in which case, we would actually deal with it correctly. Helpfully enough, we don't actually support anything but unsigned ints in multiband images, so practically we wound up doing the right thing for the wrong reasons. 

The current code just uses the `SAMPLEFORMAT` tag directly, which is fine for single band images but since we don't have any multi-item tuples defined, it's wrong for any color tiff that sets the `SAMPLEFORMAT` tag.

**Comment by wiredfool** [comment]
Try this branch: https://github.com/wiredfool/Pillow/tree/sampleformat

Is the copyleft.tiff diestibutable under an MIT like license as a test image?

(I'm somewhat worried about the war that that question could ignite...)

**Comment by pawl** [comment]
@wiredfool Works great! My example above works now. The image uploaded successfully and it fixed our tests in flask-admin too: https://i.imgur.com/l1naZYv.png

Thank you so much for your hard work on Pillow.

**Comment by wiredfool** [comment]
Cool, glad it worked.

**Comment by manakor** [comment]
I am wondering whether and when Pillow 3.0.0 is going to have this issue fixed?

Since the same issue happened to me when using `Image.open('image.tiff')` with latest Pillow, originally installed with `pip install Pillow`.  
However, all works, when installed from a repo @wiredfool pointed out:
````python
pip install git+https://github.com/wiredfool/Pillow
````

**Comment by radarhere** [comment]
The fix is included in the latest release of Pillow, 3.1.0.

At the moment, there are no planned subsequent releases of Pillow 3.0.
